src/plugins/explore/public/components/fields_selector/lib/get_details.ts

```diff
@@ -68,5 +69,12 @@ export function getDetails(
       bucket.display = indexPattern.getFormatterForField(field).convert(bucket.value);
     }
   }
+
+  // Save updated index pattern with topQueryValues to cache
+  if (indexPattern && indexPattern.id) {
```

Comment:
If you could just update to dataView. Or dataset since that is where we want to be